test/assert.js

```diff
@@ -237,3 +283,122 @@ QUnit.test( "raises, alias for throws", function( assert ) {
 		throw "my error";
 	});
 });
+
+QUnit.module( "failing assertions", {
+	beforeEach: function( assert ) {
+		assert.push = function( result, actual, expected, message ) {
+
+			// inverts the result so we can test failing assertions
+			QUnit.assert.push.call( assert, !result, actual, expected, message );
```

Comment:
Would be good to get rid of this global as well, instead saving a reference to the original `assert.push` and calling the original here.